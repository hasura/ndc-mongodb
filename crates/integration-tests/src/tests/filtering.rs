use insta::assert_yaml_snapshot;

use crate::graphql_query;

#[tokio::test]
async fn filters_on_extended_json_using_string_comparison() -> anyhow::Result<()> {
    // Skip this test in MongoDB 5 because the example fails there. We're getting an error:
    //
    // > Kind: Command failed: Error code 5491300 (Location5491300): $documents' is not allowed in user requests, labels: {}
    //
    // This doesn't affect native queries that don't use the $documents stage.
    if let Ok(image) = std::env::var("MONGODB_IMAGE") {
        if image == "mongo:5" {
            return Ok(());
        }
    }

    assert_yaml_snapshot!(
        graphql_query(
            r#"
                query Filtering {
                  extendedJsonTestData(where: { value: { _regex: "hello" } }) {
                    type
                    value
                  }
                }
            "#
        )
        .run()
        .await?
    );
    Ok(())
}
