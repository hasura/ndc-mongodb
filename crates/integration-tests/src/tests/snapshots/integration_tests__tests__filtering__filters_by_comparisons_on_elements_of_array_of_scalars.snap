---
source: crates/integration-tests/src/tests/filtering.rs
expression: "graphql_query(r#\"\n            query MyQuery {\n              movies(where: { cast: { _eq: \"Albert Austin\" } }) {\n                title\n                cast\n              }\n            }\n            \"#).run().await?"
---
data:
  movies:
    - title: The Immigrant
      cast:
        - Charles Chaplin
        - Edna Purviance
        - Eric Campbell
        - Albert Austin
errors: ~
